x/pricefeed/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,201 @@
+
+	suite.setTstPrice()
+
+	expectedPrice := types.NewCurrentPrice("tstusd", sdk.MustNewDecFromStr("0.34"))
+
+	res, err := suite.keeper.Price(sdk.WrapSDKContext(suite.ctx), &types.QueryPriceRequest{MarketId: "tstusd"})
+	suite.NoError(err)
+	// Price median of addTstPrices below
+	suite.Equal(expectedPrice, res.Price)
+
+	_, err = suite.keeper.Price(sdk.WrapSDKContext(suite.ctx), &types.QueryPriceRequest{MarketId: "invalid"})
+	suite.Equal("rpc error: code = InvalidArgument desc = invalid market ID", err.Error())
+
+	prices, err := suite.keeper.Prices(sdk.WrapSDKContext(suite.ctx), &types.QueryPricesRequest{})
+	suite.NoError(err)
+
+	// Includes previous same price
+	suite.Contains(prices.Prices, expectedPrice)
+}
+
+func (suite *grpcQueryTestSuite) TestGrpcRawPrices() {
+	params := types.NewParams([]types.Market{
+		{MarketID: "tstusd", BaseAsset: "tst", QuoteAsset: "usd", Oracles: []string{}, Active: true},
+	})
+	suite.keeper.SetParams(suite.ctx, params)
+
+	suite.setTstPrice()
+
+	res, err := suite.keeper.RawPrices(sdk.WrapSDKContext(suite.ctx), &types.QueryRawPricesRequest{MarketId: "tstusd"})
+	suite.NoError(err)
+
+	suite.Equal(3, len(res.RawPrices))
+
+	suite.ElementsMatch(
+		res.RawPrices,
+		[]types.PostedPrice{
+			types.NewPostedPrice(
+				"tstusd",
+				suite.addrs[0].String(),
+				sdk.MustNewDecFromStr("0.33"),
+				suite.now.Add(time.Hour*1),
+			),
+			types.NewPostedPrice(
+				"tstusd",
+				suite.addrs[1].String(),
+				sdk.MustNewDecFromStr("0.35"),
+				suite.now.Add(time.Hour*1),
+			),
+			types.NewPostedPrice(
+				"tstusd",
+				suite.addrs[2].String(),
+				sdk.MustNewDecFromStr("0.34"),
+				suite.now.Add(time.Hour*1),
+			),
+		},
+	)
+
+	_, err = suite.keeper.RawPrices(sdk.WrapSDKContext(suite.ctx), &types.QueryRawPricesRequest{MarketId: "invalid"})
+	suite.Equal("rpc error: code = InvalidArgument desc = invalid market ID", err.Error())
+}
+
+func (suite *grpcQueryTestSuite) TestGrpcOracles() {
+	params := types.NewParams([]types.Market{
+		{MarketID: "tstusd", BaseAsset: "tst", QuoteAsset: "usd", Oracles: []string{}, Active: true},
+	})
+	suite.keeper.SetParams(suite.ctx, params)
+
+	res, err := suite.keeper.Oracles(sdk.WrapSDKContext(suite.ctx), &types.QueryOraclesRequest{MarketId: "tstusd"})
+	suite.NoError(err)
+	suite.Empty(res.Oracles)
+
+	var oracles []string
+	for _, a := range suite.addrs {
+		oracles = append(oracles, a.String())
+	}
+
+	params = types.NewParams([]types.Market{
+		{MarketID: "tstusd", BaseAsset: "tst", QuoteAsset: "usd", Oracles: oracles, Active: true},
+	})
+	suite.keeper.SetParams(suite.ctx, params)
+
+	res, err = suite.keeper.Oracles(sdk.WrapSDKContext(suite.ctx), &types.QueryOraclesRequest{MarketId: "tstusd"})
+	suite.NoError(err)
+	suite.ElementsMatch(res.Oracles, oracles)
+
+	_, err = suite.keeper.Oracles(sdk.WrapSDKContext(suite.ctx), &types.QueryOraclesRequest{MarketId: "invalid"})
+	suite.Equal("rpc error: code = InvalidArgument desc = invalid market ID", err.Error())
```

Comment:
nit: Not finding a market feels more like a `NotFound` error code than an `InvalidArgument`